src/main/kotlin/no/nav/familie/klage/kabal/event/BehandlingEventService.kt

```diff
@@ -24,21 +29,38 @@ class BehandlingEventService(
     private val behandlingRepository: BehandlingRepository,
     private val fagsakRepository: FagsakRepository,
     private val personRepository: FagsakPersonRepository,
+    private val stegService: StegService
 ) {
 
     private val logger = LoggerFactory.getLogger(javaClass)
 
     fun handleEvent(behandlingEvent: BehandlingEvent) {
-        opprettOppgave(behandlingEvent)
-        ferdigstillKlagebehandling()
-    }
-
-    private fun opprettOppgave(behandlingEvent: BehandlingEvent) {
-
         val behandling = behandlingRepository.findByEksternBehandlingIdAndFagsystem(
             behandlingEvent.kildeReferanse,
             behandlingEvent.kilde
         )
+
+        when (behandlingEvent.type) {
+            BehandlingEventType.KLAGEBEHANDLING_AVSLUTTET -> behandleKlageAvsluttet(behandling, behandlingEvent)
+            else -> behandleAnke(behandling, behandlingEvent)
+        }
+    }
+
+    private fun behandleAnke(behandling: Behandling, behandlingEvent: BehandlingEvent) {
+        opprettOppgave(behandlingEvent, behandling)
+    }
+
+    private fun behandleKlageAvsluttet(behandling: Behandling, behandlingEvent: BehandlingEvent) {
+        when (behandling.status) {
+            BehandlingStatus.FERDIGSTILT -> logger.error("Mottatt event på ferdigstilt behandling $behandlingEvent") // TODO korrigeringer - kan vi få det?
```

Comment:
Bør vi vurdere å kaste feil hvis vi ikke forventer denne situasjonen?

Reply:
Denne prøver å løse: håndtere å lese samme melding to ganger uten å feile. 

Vi har diskutert den litt og klarte ikke lande noe helt. warn var også på bordet. 

Mulig en sjekk på "FERDIGSTILLT" ikke er nok. Vi er fortsatt i tenkeboks her (dekkes av annen favro oppgave checkboks).